lib/humanize/module.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+require 'bigdecimal'
+require_relative './locales'
+
+module Humanize
+  # Big numbers are big: http://wiki.answers.com/Q/What_number_is_after_vigintillion&src=ansTT
+
+  class << self
+    def format(number,
+               locale: Humanize.config.default_locale,
+               decimals_as: Humanize.config.decimals_as)
+      locale_class, spacer = Humanize.for_locale(locale)
+  
+      return locale_class::SUB_ONE_GROUPING[0] if number.zero?
+  
+      infinity = number.to_f.infinite?
+      if infinity
+        infinity_word = locale_class::INFINITY
+        return infinity == 1 ? infinity_word : "#{locale_class::NEGATIVE}#{spacer}#{infinity_word}"
+      elsif number.is_a?(Float) && number.nan?
+        return locale_class::UNDEFINED
+      end
+  
+      sign = locale_class::NEGATIVE if number.negative?
+  
+      parts = locale_class.new.humanize(number.abs)
+      Humanize.process_decimals(number, locale_class, locale, parts, decimals_as, spacer)
+      Humanize.stringify(parts, sign, spacer)
+    end
+  
+    def for_locale(locale)
+      case locale.to_sym
+      # NOTE: add locales here in ealphabetical order
+      when :az, :de, :en, :es, :fr, :id, :ms, :pt, :ru, :vi
+        [Object.const_get("Humanize::#{locale.capitalize}"), ' ']
+      when :th
+        [Humanize::Th, '']
+      when :tr
+        [Humanize::Tr, ' ']
+      when :jp
+        [Humanize::Jp, '']
```

Comment:
This is missing the recent zh-TW addition.

Reply:
@radar Oops, I didn't mean to make any changes to this locale list. I just looked over in `master` and didn't find any reference to `zh`. Can you point me to it?

And I'm looking at the build failures.

Reply:
The build failures were rubocop suggestions. All addressed here: https://github.com/radar/humanize/pull/88/commits/0abcfc97e926fbc5c9d0a6a93f573ad3d362ea82

Reply:
> @radar Oops, I didn't mean to make any changes to this locale list. I just looked over in `master` and didn't find any reference to `zh`. Can you point me to it?
> 
> And I'm looking at the build failures.

Getting ahead of myself. It's in https://github.com/radar/humanize/pull/87.

Reply:
Ok, merged #87. So can we now get that in here? I think it's just the two lines for the when + the config.

Reply:
Ok, I've merged latest `master` with that PR here.